pkg/kubectl/cmd/wait/wait.go

```diff
@@ -221,14 +227,24 @@ func (o *WaitOptions) RunWait() error {
 		}
 		return err
 	})
+
+	if err != nil {
```

Comment:
You can remove this check and just "return err" at the end of the function.

Reply:
You can also remove the empty lines. :)

Reply:
> You can remove this check and just "return err" at the end of the function.

Hm. I think, if I remove this condition it will break the existing behaviour. For example 

```
kubectl wait deployment something-that-does-not-exist --for condition=available --timeout=5s
```

is currently printing an error from the API:

```
Error from server (NotFound): deployments.extensions "something-that-does-not-exist" not found
```

with the change you suggest to make it will printing "error: failed to find matching resources" in this case.

Another example (invalid command args):
```
kubectl wait deployment --for condition=available --timeout=5s
```

currently provides a helpful message: "error: resource(s) were provided, but no name, label selector, or --all flag specified", but if I remove the check it will also print "error: failed to find matching resources".

So, I think, we need return errors from the visitors first.